packages/SwingSet/src/kernel/vatManager/supervisor-subprocess-xsnap.js

```diff
@@ -0,0 +1,237 @@
+   */
+  async function setBundle(
+    vatID,
+    bundle,
+    vatParameters,
+    virtualObjectCacheSize,
+  ) {
+    /** @type { (item: Tagged) => unknown } */
+    function doSyscall(vatSyscallObject) {
+      return JSON.parse(
+        decoder.decode(
+          // @ts-ignore
+          // eslint-disable-next-line no-undef
+          issueCommand(Item.encode(['syscall', ...vatSyscallObject])),
+        ),
+      );
+    }
+
+    const syscall = harden({
+      send: (...args) => doSyscall(['send', ...args]),
+      callNow: (...args) => doSyscall(['callNow', ...args]),
+      subscribe: (...args) => doSyscall(['subscribe', ...args]),
+      fulfillToData: (...args) => doSyscall(['fulfillToData', ...args]),
+      fulfillToPresence: (...args) => doSyscall(['fulfillToPresence', ...args]),
+      reject: (...args) => doSyscall(['reject', ...args]),
+    });
+
+    const vatPowers = {
+      Remotable,
+      getInterfaceOf,
+      makeMarshal,
+      testLog,
+    };
+
+    const ls = makeLiveSlots(
+      syscall,
+      vatID,
+      vatPowers,
+      vatParameters,
+      // @ts-ignore  TODO: defend against non-numeric?
+      virtualObjectCacheSize,
+      // TODO: lsgc? API drift?
+    );
+
+    const endowments = {
+      ...ls.vatGlobals,
+      console: makeConsole(`SwingSet:vatWorker`),
+      assert,
+      // @ts-ignore bootstrap provides HandledPromise
+      // eslint-disable-next-line no-undef
+      HandledPromise,
+    };
+    const vatNS = await importBundle(bundle, { endowments });
+    workerLog(`got vatNS:`, Object.keys(vatNS).join(','));
+    ls.setBuildRootObject(vatNS.buildRootObject);
+    dispatch = ls.dispatch;
+    assert(dispatch);
+    workerLog(`got dispatch:`, Object.keys(dispatch).join(','));
+    return ['dispatchReady'];
+  }
+
+  /** @type { (item: Tagged) => Promise<Tagged> } */
+  async function handleItem([tag, ...args]) {
+    switch (tag) {
+      case 'setBundle':
+        assert(!dispatch, 'cannot setBundle again');
+        return setBundle(args[0], args[1], args[2], args[3]);
+      case 'deliver': {
+        assert(dispatch, 'cannot deliver before setBundle');
+        const [dtype, ...dargs] = args;
+        switch (dtype) {
+          case 'message': {
+            const res = await doMessage(dargs[0], dargs[1]);
+            return ['ok', ...res];
+          }
+          case 'notify': {
+            const res = await doNotify(dargs[0], dargs[1]);
+            return ['ok', ...res];
+          }
+          default:
+            throw Error(`bad delivery type ${dtype}`);
+        }
+      }
+      default:
+        workerLog('handleItem: bad tag', tag, args.length);
+        return ['bad tag', tag];
+    }
+  }
+
+  let seq = 0;
+  /** @type { Tagged? } */
+  let result = null;
+
+  /** @type { (item: Tagged) => Tagged } */
+  function answerItem(item) {
+    const [tag, ...args] = item;
+    workerLog('answerItem', tag, args.length);
+    if (tag === 'getResult') {
```

Comment:
per our discussion yesterday, I think it'd be slightly cleaner for the command that provokes a delivery to return the results of that delivery itself, rather than requiring the kernel process to send a second command to fetch the results of the previous one, but see what that does to the implementation and build whatever seems best